presto-main/src/main/java/com/facebook/presto/server/security/KerberosAuthenticator.java

```diff
@@ -64,7 +64,8 @@ public KerberosAuthenticator(KerberosConfig config)
         System.setProperty("java.security.krb5.conf", config.getKerberosConfig().getAbsolutePath());
 
         try {
-            String hostname = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase(Locale.US);
+            String canonicalLocalHostName = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase(Locale.US);
+            String hostname = config.getServiceHostName() == null ? canonicalLocalHostName : config.getServiceHostName();
```

Comment:
This can simplify to (using `com.google.common.base.MoreObjects`)
```java
String hostname = firstNonNull(config.getServiceHostName(), canonicalLocalHostName);

Reply:
1. we force `InetAddress.getLocalHost().getCanonicalHostName()` into lowercase. Can `config.getServiceHostName()` be set to non-lowercase?
2. is there any situation that `InetAddress.getLocalHost().getCanonicalHostName()` could fail? If so, we should avoid running the code when `config.getServiceHostName()` is set

Reply:
I've updated the commit, trying to address the tree comments:
0. better readability
1. enforce lowercase for config.getServiceHostName()
2. Lazy call of InetAddress.getLocalHost()

Since InetAddress.getLocalHost() throws I didn't succeed to have a clean combination like
`String hostname = Optional.ofNullable(config.getServiceHostName())
                    .orElseGet(() -> InetAddress.getLocalHost().getCanonicalHostName())
                    .toLowerCase(Locale.US);`

So I think the update solve 1 & 2, not so much the readability

Thanks a lot for your reviews.

Reply:
commit updated to follow this advice